### PR TITLE
docs: [el-upload]add example of setting on-exceed to cover the old file

### LINKED ISSUE #4655: [Feature Request] Automatically replace current file in el-upload
### Existing Component
Yes

### Component Name
el-upload

### Description
Currently, when using el-upload with multiple=false and auto-upload=false, you need to click the cross to remove the currently imported file first before importing a new file. And there is no error/warning when you try to upload another, it just does nothing. There seem to be 2 solutions:
1. Either disable the upload button when multiple=false and a file is already being imported
2. Or automatically replace the previous file when upload anew (without having to remove it first)

Our UX design recommends 2nd choice but I have not look into feasibility to be honest.

### PR DESCRIPTION
**Pull request by Alanscut**
Add a example of setting `limit` and `on-exceed` to automatically replace the previous file when select a new file
Add guidance on how to use  `clearFiles()`
fix: #4655 
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by emojiiii** [comment]
I don't think I need this example. This is a special scenario

File: docs/en-US/component/upload.md [hunk 1]
```diff
@@ -123,8 +131,9 @@ upload/manual
...
-| ------------ | --------------------------------------------------------------------------------------- | --------------------------- |
-| clearFiles   | clear the uploaded file list (this method is not supported in the `before-upload` hook) | —                           |
-| abort        | cancel upload request                                                                   | （ file: fileList's item ） |
-| submit       | upload the file list manually                                                           | —                           |
+| Methods Name | Description                                                                     | Parameters                                                                    | Default             |
+| ------------ | ------------------------------------------------------------------------------- | ----------------------------------------------------------------------------- | ------------------- |
+| clearFiles   | clear the file list (this method is not supported in the `before-upload` hook). | UploadStatus[] (UploadStatus = 'ready' \| 'uploading' \| 'success' \| 'fail') | ['success', 'fail'] |
```
Comment (btea) [inline docs/en-US/component/upload.md:136]: The default value should be `['success', 'fail', 'ready', 'uploading']`  rel #4365

Comment (Alanscut) [reply docs/en-US/component/upload.md:136]: ok

**Comment by Alanscut** [comment]
@emojiiii 
I don't think this requirement is very special.
In other words, adding this example to give users one more choice, there is no loss

File: packages/components/upload/src/useHandlers.ts [hunk 2]
```diff
@@ -34,7 +34,9 @@ export default (props: IUseHandlersProps) => {
     uploadRef.value.abort(file)
   }
 
-  function clearFiles(status: UploadStatus[] = ['success', 'fail']) {
+  function clearFiles(
+    status: UploadStatus[] = ['ready', 'uploading', 'success', 'fail']
+  ) {
     uploadFiles.value = uploadFiles.value.filter((row) => {
       return status.indexOf(row.status) === -1
```
Comment (btea) [inline packages/components/upload/src/useHandlers.ts:41]:
```suggestion
      return !status.includes(row.status)
```
I'll close #4365

Comment (Alanscut) [reply packages/components/upload/src/useHandlers.ts]: ok

File: docs/examples/upload/limit-cover.vue [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    </template>
+  </el-upload>
+</template>
+
+<script setup lang="ts">
+import { ref } from 'vue'
+const upload = ref()
+
+const handleExceed = (files) => {
+  upload.value.clearFiles(['ready', 'success', 'fail', 'uploading'])
```
Comment (btea) [inline docs/examples/upload/limit-cover.vue:33]: The default parameter can be omitted 
```suggestion
  upload.value.clearFiles()
```

**Review by btea — APPROVED** [review]

**Pull request merged**